src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -201,14 +198,19 @@ mkldnn_memory_format_t GetDefaultFormat(mkldnn::memory::desc desc) {
   } else if (desc.data.ndims == 4) {
     switch (desc.data.format) {
       case mkldnn_nchw:
-      case mkldnn_nhwc:
-      case mkldnn_chwn:
       case mkldnn_nChw8c:
       case mkldnn_nChw16c:
         return mkldnn_nchw;
+      case mkldnn_nhwc:
+        return mkldnn_nhwc;
+      case mkldnn_chwn:
+        return mkldnn_chwn;
```

Comment:
Seems I lack some of background here. So mxnet has several different **default** layouts for 4D image data? How does NDArray tell the difference of nchw and nhwc?